app/src/main/java/org/mozilla/fenix/gleanplum/MessagesManager.kt

```diff
@@ -0,0 +1,94 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.gleanplum
+
+import android.content.Context
+import org.mozilla.experiments.nimbus.internal.FeatureHolder
+import org.mozilla.fenix.nimbus.Messaging
+import org.mozilla.fenix.nimbus.StyleData
+import java.util.SortedSet
+import kotlin.Comparator
+
+class MessagesManager(
+    private val context: Context,
+    private val storage: MessageStorage,
+    private val messagingFeature: FeatureHolder<Messaging>
+) : MessageController {
+
+    private val availableMessages: SortedSet<Message> = sortedSetOf(
+        Comparator { message1, message2 -> message2.style.priority.compareTo(message1.style.priority) }
+    )
+
+    fun areMessagesAvailable(): Boolean {
+        return availableMessages.isNotEmpty()
+    }
+
+    override fun getNextMessage(): Message? {
+        return availableMessages.first()
```

Comment:
I just noticed this was a draft, but typed too much to delete :(

 * run the JEXL triggers here.
 * [handle control messages and exposure events][0]

[0]:  https://docs.google.com/document/d/1zOdYpXkunWm-0MeUjTjuS_CmbfpJrsAta0vxgzEzy14/edit